src/main/scala/org/allenai/plugins/VersionInjectorPlugin.scala

```diff
@@ -117,6 +144,45 @@ object VersionInjectorPlugin extends AutoPlugin {
         gitConfFile
     }
 
+  val injectCacheKeyTask =
+    injectCacheKey <<= (resourceManaged in Compile, organization, name, version, streams, cacheKey) map {
+      // scalastyle: ignore
+      (resourceManaged, org, name, version, s, cacheKeyResult) =>
+        val cacheKeyConfFile = resourceManaged / org / cleanArtifactName(name) / "cacheKey.conf"
+        val cacheKeyContents = "cacheKey: " + "\"" + cacheKeyResult + "\""
+        IO.write(cacheKeyConfFile, cacheKeyContents)
+        cacheKeyConfFile
+    }
+
+  lazy val gitMRCs = Def.taskDyn {
+    val MRCs = buildDependencies.value.classpathRefs(thisProjectRef.value)
+
+    val filter = ScopeFilter(inProjects(MRCs: _*))
+    gitMRC.all(filter)
+  }
+
+  val cacheKeyTask = cacheKey := {
+    import java.io.{ File => JFile }
+
+    def getFileList(dir: String): Seq[JFile] = new JFile(dir) match {
+      case f if f.exists && f.isDirectory => f.listFiles.filter(_.isFile).toSeq
+      case _ => Seq.empty
+    }
+    val stageDir = baseDirectory.value + "/target/universal/stage"
+    val allFiles: Seq[JFile] = Seq("bin", "conf", "lib", "public")
+      .map { dir => stageDir + "/" + dir }
+      .flatMap(getFileList)
+
+    val (filesToGetGitSha, filesToHash) = allFiles partition { f: JFile =>
+      val fileName = f.getName
+      fileName.startsWith("org.allenai.ari-api") ||
+        fileName.startsWith("org.allenai.solvers-") ||
+        fileName.startsWith("org.allenai.ari-solvers-")
+    }
+    val hashes = filesToHash.map(Hash.apply).map(Hash.toHex).mkString
```

Comment:
This still relies on filesystem and SBT subproject ordering, both of which could fluctuate. You should put all hash values into a set, sort them, then take the final hash value.